### PR TITLE
ref(snql) Add a migration to recreate subscriptions in SnQL

### PR DESCRIPTION
**Pull request by evanh**
This migration will recreate all the existing subscriptions in Snuba. New
subscriptions are created with SnQL, so this effectively rebuilds all existing
subscriptions with SnQL.

This is marked as dangerous because it has already been run manually in prod and
there's no point in running it again. This has been added so that we can
deprecate the Snuba legacy API in later releases.

### CODE REVIEW

**Review by wedamija — APPROVED** [review]
lgtm, this is just to make on-prem and ST in sync with getsentry?

**Comment by evanh** [comment]
> lgtm, this is just to make on-prem and ST in sync with getsentry?

@wedamija Yes exactly. This script has already been run in prod.

File: src/sentry/migrations/0233_recreate_subscriptions_in_snuba.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                logging.exception(f"failed to recreate {subscription.subscription_id}: {e}")
+                continue
+
+            try:
+                _delete_from_snuba(
+                    QueryDatasets(subscription.snuba_query.dataset),
+                    subscription.subscription_id,
+                )
+            except Exception as e:
+                logging.exception(f"failed to delete {subscription.subscription_id}: {e}")
```
Comment (nikhars) [inline src/sentry/migrations/0233_recreate_subscriptions_in_snuba.py:32]: Is it possible that creating new subscription at line 21 succeeded above but deleting the old subscription failed. If yes, wouldn't you have 2 subscriptions in that case. Should you delete the subscription created at line 21 over here then?

Comment (evanh) [reply src/sentry/migrations/0233_recreate_subscriptions_in_snuba.py:32]: Yeah that's a good idea.

**Pull request merged**